format/wav/Reader.hx

```diff
@@ -100,15 +100,38 @@ class Reader {
 		// data
 		if (nextChunk != "data")
 			throw "expected data subchunk";
-		
+
 		var datalen = readInt();
-		
-		// Some files report an incorrect length, so we'll
-		// read the whole file, then subtract if necessary
-		var data = i.readAll ();
-		if (data.length > datalen) 
-			data = data.sub (0, datalen);
-		
+
+		var data = i.read(datalen);
+
+		var cuePoints = new Array<CuePoint>();
+		try {
+			var nextChunk = i.readString (4);
+			while (nextChunk != "cue ") {
+				// read past other subchunks
+				i.read(readInt());
+				nextChunk = i.readString (4);
+			}
+
+			// cue
+			if (nextChunk == "cue ") {
+				readInt();

```

Comment:
Another way to write this:

while( true ) {
    var chunk = i.readString(4);
    switch( chunk ) {
    case "cue ": 
       // decode cue data
      ...
      ...
    default:
          i.read(readInt()); // skip chunk data
    }
}